src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoTreeNodes.cs

```diff
@@ -951,6 +1026,55 @@ protected override void OnExpandPopulateFolders(IList<TreeNode> currentChildren,
                 NodeTypeId = NodeTypes.Tables,
                 IsSystemObject = false,
                 SortPriority = SmoTreeNode.NextSortPriority,
+        		FilterProperties = new NodeFilterProperty[]
```

Comment:
>FilterProperties = new NodeFilterProperty[] [](http://example.com/codeflow?start=10&length=43)

is this list identical to the one starting on line 790? If so just move the duped code into a function

Reply:
oh it's template generated code.
I am ripping out template generated code from SMO lol

Reply:
Yeah, it is generated by https://github.com/microsoft/sqltoolsservice/blob/fcaca65a14b4570309be839d2aa933c117109c47/src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoTreeNodes.tt

Reply:
I'm starting with the collection classes because they were built for .Net 2 and rely too much on reflection and need to be modernized.
I'm not eliminating codegen entirely, but I'm replacing the collection templates with a C# source generator whose output is only about 10 lines of code per collection instead of dozens. 90% of the code output by the template I moved into the base classes by re-implementing them as generics.

Reply:
if you ever want to replace t4 with your own source generator, take a look at [https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.md](https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.md)

Reply:
This is great. I will try to convert the tt files in c# code. Will be easier to work and debug with that.